package/android/cpp/rnskia-android/SkiaOpenGLHelper.h

```diff
@@ -0,0 +1,246 @@
+#include "include/gpu/GrDirectContext.h"
+#include "include/gpu/gl/GrGLInterface.h"
+
+#pragma clang diagnostic pop
+
+namespace RNSkia {
+struct SkiaOpenGLContext {
+  SkiaOpenGLContext() {
+    glDisplay = EGL_NO_DISPLAY;
+    glContext = EGL_NO_CONTEXT;
+    gl1x1Surface = EGL_NO_SURFACE;
+    glConfig = 0;
+    directContext = nullptr;
+  }
+  EGLContext glContext;
+  EGLDisplay glDisplay;
+  EGLConfig glConfig;
+  EGLSurface gl1x1Surface;
+  sk_sp<GrDirectContext> directContext;
+};
+
+class SkiaOpenGLHelper {
+public:
+  /**
+   * Calls eglMakeCurrent on the surface provided using the provided
+   * thread context.
+   * @param context Skia OpenGL context to use
+   * @param surface Surface to set as current
+   * @return true if eglMakeCurrent was successfull
+   */
+  static bool makeCurrent(SkiaOpenGLContext *context, EGLSurface glSurface) {
+    // We don't need to call make current if we already are current:
+    if (eglGetCurrentDisplay() != context->glDisplay ||
+        eglGetCurrentSurface(EGL_DRAW) != glSurface ||
+        eglGetCurrentSurface(EGL_READ) != glSurface ||
+        eglGetCurrentContext() != context->glContext) {
+
+      // Make current!
+      if (eglMakeCurrent(context->glDisplay, glSurface, glSurface,
+                         context->glContext) != EGL_TRUE) {
+        RNSkLogger::logToConsole("eglMakeCurrent failed: %d\n", eglGetError());
+        return false;
+      }
+      return true;
+    }
+    return true;
+  }
+
+  /**
+   * Calls the eglSwapBuffer in the current thread with the provided surface
+   * @param context Thread context
+   * @param glSurface surface to present
+   * @return true if eglSwapBuffers succeeded.
+   */
+  static bool swapBuffers(SkiaOpenGLContext *context, EGLSurface glSurface) {
+    if (eglSwapBuffers(context->glDisplay, glSurface) != EGL_TRUE) {
+      RNSkLogger::logToConsole("eglSwapBuffers failed: %d\n", eglGetError());
+      return false;
+    }
+    return true;
+  }
+
+  /***
+   * Creates a new Skia direct context backed by the provided eglContext in the
+   * SkiaOpenGLContext.
+   * @param context Context to store results in
+   * @param sharedContext Shared Context
+   * @return true if the call to create a skia direct context suceeded.
+   */
+  static bool
+  createSkiaDirectContext(SkiaOpenGLContext *context,
+                          std::atomic<EGLContext> *sharedEglContext) {
+    // Initialize OpenGL
+    if (!initializeOpenGL(context)) {
+      return false;
+    }
+
+    // Create the OpenGL context
+    if (!createOpenGLContext(context, sharedEglContext)) {
+      return false;
+    }
+
+    if (context->directContext == nullptr) {
+      // Create attributes for a simple 1x1 pbuffer surface that we can
+      // use to activate and create Skia direct context for
+      const EGLint offScreenSurfaceAttribs[] = {EGL_WIDTH, 1, EGL_HEIGHT, 1,
+                                                EGL_NONE};
+
+      context->gl1x1Surface = eglCreatePbufferSurface(
```

Comment:
does this has a corresponding eglDestroySurface?

Reply:
Nope, I'll create some destructors for this to make this symetric.